lib/child_process.js

```diff
@@ -245,39 +252,45 @@ exports.execFile = function(file /*, args, options, callback*/) {
   }
 
   if (child.stdout) {
-    if (encoding)
-      child.stdout.setEncoding(encoding);
+    stdoutState = child.stdout._readableState;
 
     child.stdout.addListener('data', function(chunk) {
-      stdoutLen += chunk.length;
+      // If `child.stdout.setEncoding('utf8') happened in userland, convert
+      // string to Buffer and save encoding for later.
```

Comment:
The last part of this comment doesn't make sense. The string's encoding is not being saved anywhere for later.